From 33cf77695e9120903f0aefa7b53cf5e060d03b44 Mon Sep 17 00:00:00 2001
From: Murali Karicheri <m-karicheri2@ti.com>
Date: Fri, 4 Aug 2017 18:15:52 -0400
Subject: [PATCH 46/50] prp: remove NETIF_F_HW_L2FW_DOFFLOAD requirement

For PRP offload, there is no L2 offload happening at the lower layer
firmware or hardware. So remove this dependency from the drivers.

Signed-off-by: Murali Karicheri <m-karicheri2@ti.com>
---
 net/hsr-prp/hsr_prp_device.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/net/hsr-prp/hsr_prp_device.c b/net/hsr-prp/hsr_prp_device.c
index 4e84cb1..cb58a7a 100644
--- a/net/hsr-prp/hsr_prp_device.c
+++ b/net/hsr-prp/hsr_prp_device.c
@@ -546,9 +546,10 @@ int hsr_prp_dev_finalize(struct net_device *hsr_prp_dev,
 	     (slave[1]->features & NETIF_F_HW_PRP_RX_OFFLOAD)))
 		priv->rx_offloaded = true;
 
-	/* HSR LRE L2 forward offload supported in lower device? */
-	if ((slave[0]->features & NETIF_F_HW_L2FW_DOFFLOAD) &&
-	    (slave[1]->features & NETIF_F_HW_L2FW_DOFFLOAD))
+	/* HSR LRE L2 forward offload supported in lower device for hsr? */
+	if ((priv->prot_version < PRP_V1) &&
+	    ((slave[0]->features & NETIF_F_HW_L2FW_DOFFLOAD) &&
+	     (slave[1]->features & NETIF_F_HW_L2FW_DOFFLOAD)))
 		priv->l2_fwd_offloaded = true;
 
 	res = register_netdevice(hsr_prp_dev);
-- 
1.9.1

